itests/hive-unit/src/test/java/org/apache/hadoop/hive/ql/parse/TestStatsReplicationScenarios.java

```diff
@@ -301,12 +338,106 @@ private String dumpLoadVerify(List<String> tableNames, String lastReplicationId,
     return dumpTuple.lastReplicationId;
   }
 
+  /**
+   * Run a bootstrap that will fail.
+   * @param tuple the location of bootstrap dump
+   */
+  private void failBootstrapLoad(WarehouseInstance.Tuple tuple, int failAfterNumTables) throws Throwable {
+    // fail setting ckpt directory property for the second table so that we test the case when
+    // bootstrap load fails after some but not all tables are loaded.
+    BehaviourInjection<CallerArguments, Boolean> callerVerifier
+            = new BehaviourInjection<CallerArguments, Boolean>() {
+      int cntTables = 0;
+      @Nullable
+      @Override
+      public Boolean apply(@Nullable CallerArguments args) {
+        cntTables++;
+        if (args.dbName.equalsIgnoreCase(replicatedDbName) && cntTables > failAfterNumTables) {
+          injectionPathCalled = true;
+          LOG.warn("Verifier - DB : " + args.dbName + " TABLE : " + args.tblName);
+          return false;
+        }
+        return true;
+      }
+    };
+
+    InjectableBehaviourObjectStore.setAlterTableModifier(callerVerifier);
+    try {
+      replica.loadFailure(replicatedDbName, tuple.dumpLocation);
+      callerVerifier.assertInjectionsPerformed(true, false);
+    } finally {
+      InjectableBehaviourObjectStore.resetAlterTableModifier();
+    }
```

Comment:
Shall add necessary validations to see only so many tables are loaded at this point of time.

Reply:
I don't think we need to be really hard and fast about the exact number of tables loaded. All we are testing is whether there was a failure and the retry loaded the stats successfully. Current set of checks is enough for that.

Reply:
Ok